symfit/core/minimizers.py

```diff
@@ -358,11 +412,86 @@ class ScipyGradientMinimize(ScipyMinimize, GradientMinimizer):
     """
     Base class for :func:`scipy.optimize.minimize`'s gradient-minimizers.
     """
-    def __init__(self, *args, **kwargs):
-        super(ScipyGradientMinimize, self).__init__(*args, **kwargs)
+    @keywordonly(jacobian=None)
+    def execute(self, **minimize_options):
+        if 'jacobian' in minimize_options:
+            jacobian = minimize_options.pop('jacobian')
+        else:
+            jacobian = self.wrapped_jacobian
+        return super(ScipyGradientMinimize, self).execute(jacobian=jacobian, **minimize_options)
```

Comment:
For the ScipyHessianMinimizer the user needs to be able to override the jacobian/hessian calculation.

Reply:
I think this is part of a more general problem: when the user wants to provide a different value to an input which symfit already defines by default, we have the potential for conflict. I think this is a problem we should solve generally, but which to my mind does not need to be solved immediately because in symfit we can typically provide the exact Hessian from the model, and in other cases the scipy default will do just fine. Unless I'm missing why this particular algorithm would be way more sensitive to this.

Reply:
In a nutshell: the scipy default are not sufficient. If you pass not jacobian/hessian information to trust-constr it will blow up into your face saying it needs both.

In my view you should be able to use any of our wrapped minimizers with any kind of model, even those for which we don't know a jacobian/hessian analytically. That means that in those cases you absolutely must use a gradient/hessian based minimizer you need to be able to provide a jacobian/hessian. In that case the responsibility is on the user to 1) make sure they actually do, and 2) whatever they pass conforms to the API and 3) produces the correct values though.

In any case, if the user thinks they know better than what we provide for them, we should honor that since they have more intimate knowledge/information about their system/model. It's up to them to analyze the ashes once it blows up though.

So to wrap up, this is not an issue in 95% of the cases where the sane symfit determined analytical jacobian/hessian are passed. Allowing users to override our defaults enables them to also use this is the other 5% of the cases though.